fuel-vm/src/interpreter/alu/wideint.rs

```diff
@@ -0,0 +1,365 @@
+
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    *dest = [<cmp_ $t:lower>](lhs, rhs, args.mode);
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_op_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    args: MathArgs,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    // LHS argument is always indirect, load it
+                    let lhs: $t = $t::from_be_bytes(read_bytes(&self.memory, b)?);
+
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    let (wrapped, overflow) = [<op_overflowing_ $t:lower>](lhs, rhs, args);
+
+                    if overflow && !is_wrapping(flag.into()) {
+                        return Err(PanicReason::ArithmeticOverflow.into());
+                    }
+
+                    *of = overflow as Word;
+                    *err = 0;
+
+                    write_bytes(&mut self.memory, owner_regs, dest_addr, wrapped.to_be_bytes())?;
+
+                    inc_pc(pc)
+                }
+
+                pub(crate) fn [<alu_wideint_mul_ $t:lower>](
+                    &mut self,
+                    dest_addr: Word,
+                    b: Word,
+                    c: Word,
+                    args: MulArgs,
+                ) -> Result<(), RuntimeError> {
+                    let owner_regs = self.ownership_registers();
+                    let (SystemRegisters { flag, mut of, mut err, pc, .. }, _) = split_registers(&mut self.registers);
+
+                    // LHS is only indirect if the flag is set
+                    let lhs: $t = if args.indirect_lhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, b)?)
+                    } else {
+                        b.into()
+                    };
+                    // RHS is only indirect if the flag is set
+                    let rhs: $t = if args.indirect_rhs {
+                        $t::from_be_bytes(read_bytes(&self.memory, c)?)
+                    } else {
+                        c.into()
+                    };
+
+                    let (wrapped, overflow) = $t::overflowing_mul(lhs, rhs);
+
+                    if overflow && !is_wrapping(flag.into()) {
```

Comment:
should not panic if `F_UNSAFEMATH` is set

> If the [F_UNSAFEMATH](https://github.com/FuelLabs/fuel-vm/pull/424/index.md#flags) flag is set, an ALU operation that would have panicked will instead set $err to true.

https://github.com/FuelLabs/fuel-specs/blob/813faf0b94bc45cd5e9b55826b57831d27bed394/src/vm/instruction_set.md?plain=1#L144

Reply:
See below